modules/auth/proc-adding-google-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,59 @@
+[id='proc-adding-google-as-an-authentication-provider_{context}']
+= Enabling the Google authentication provider
+
+{product} includes a Google authentication provider that can authenticate users by using Google OAuth2.
+
+.Procedure
+. To allow {product-short} to authenticate with Google, create OAuth credentials in Google.
+For more information, see the official Google Cloud documentation for link:https://developers.google.com/identity/openid-connect/openid-connect#appsetup[Setting up OAuth 2.0].
++
+.. To authorize {product-short} to access the user consent screen, enter the necessary information on the https://console.cloud.google.com/apis/credentials/consent/edit[*Google Cloud > Oauth consent screen > Edit app registration*] page, including the following:
++
+... On the *OAuth consent screen* tab, enter your {product-short} domain (_<APP_FQDN>_) into the *Authorized domains* field.
+... On the *Scopes* tab, select the following scopes:
++
+openid:: Associate you with your personal info on Google.
+auth/userinfo.email:: See your primary Google Account email address.
+auth/userinfo.profile:: See your personal info, including any personal info you've made publicly available.
+... On the *Test users* tab, add yourself as a test user, if using the External user type.
+
+.. To create your OAuth *Client ID* and *Client secret*, on the link:https://console.cloud.google.com/apis/credentials[*APIs & Services > Credentials*] page, click *Create Credentials*, choose *OAuth client ID*, and enter the necessary informations including the following:
+... In the *Application type* dropdown, select `Web Application`.
+... In the *Authorized JavaScript origins* section, enter the {product-short} URI: `pass:c,a,q[https://_<APP_FQDN>_]`.
+... In the *Authorized redirect URIs* section, enter the backend authentication URI set in {product-short}:
+`pass:c,a,q[https://_<APP_FQDN>_/api/auth/google/handler/frame]`.
+... Save the *Client ID* and *Client secret* for the next step.
+
+. Add your Google credentials in your {product-short} secrets.
++
+Edit your {product-short} secrets, such as `secrets-rhdh`.
++
+Add the following key/value pairs:
++
+`AUTH_GOOGLE_CLIENT_ID`:: Enter the *Client ID* that you generated on Google.
+`AUTH_GOOGLE_CLIENT_SECRET`:: Enter the *Client secret* that you generated on Google.
+
+. Set up the Google authentication provider in your {product-short} custom configuration.
```

Comment:
@themr0c @linfraze After configuring the Google Auth provider in app-config, logging in might result in a "user not found" error. To resolve this, the following configuration is required:

app-config.yaml
```
catalog:
      locations:
        - type: url
          target: https://github.com/janus-qe/rhdh-test/blob/main/user.yml
          rules:
            - allow: [User]
```
user.yml
```
apiVersion: backstage.io/v1alpha1
kind: User
metadata:
  name: rhdhtest
spec:
  memberOf: []
```

Here, name: rhdhtest is user ID of the Google account.

Reply:
@subhashkhileri thanks for letting us know! Is this step required or is it a troubleshooting step? If not required to complete the task, it might be better suited for a troubleshooting sub-section. If that is the case, we can address this with a separate Jira ticket and determine the best format and placement for the troubleshooting steps.

Reply:
@linfraze This step is required.

Reply:
@subhashkhileri The part about _logging in might result in a "user not found" error. To resolve this, the following configuration is required_? Seemed like that configuration is only required if there is an error, which would make it a troubleshooting step. But if this login error will always occur and the step is truly required to prevent it, then we can incorporate this into the main procedure.

Reply:
@Zaperex  can you please confirm if its always required or not?

Reply:
I am not sure how to write it to make it scale for enterprise customers who have more than one user. 

How to declare a user list rather than just one user?

Is it possible to declare the user list as a Kubernetes object rather than using a file on GitHub?

Reply:
@kadel Do you have any hint?

Reply:
This is actually how backstage works. User that are logging-in needs to have an existing entity in catalog. But this is not specific to Google auth, the same applies for any auth provider. Maybe we should mention that somewhere in generic auth section. 


Users are regular catalog entities, so they can be imported from any file, or there are even plugins that can import users from other systems like GitHub Orgs or Keycloak. I'm not aware of any importer plugins for Google.


The way how authenticated users are mapped to catalog users can by customized by sign-in resolvers (https://backstage.io/docs/auth/identity-resolver#sign-in-resolvers)

Reply:
I understand we have a prerequisite, that is the same for every provider, and that we could document in another place, is that correct:

In this doc in the Prerequsites section:

```

* You have provisioned all users in the {product} catalog.

```

In another doc:

```
# Provisioning users in the {product} catalog

Signing in a user into {product} requires a mapping of the user identity from the third-party auth provider to a {product-short} user identity. 

A {product-short} user identity consists in: 
- A user entity reference.
- A set of ownership references.

.Procedure

. Define the user entiry reference. TBD

. Define set of ownership references. TBD

.Verification

. Get the user list. TBD
. Get the ownership reference. TBD

```

Did I understand well?